packages/rules-unit-testing/src/impl/discovery.ts

```diff
@@ -84,29 +84,20 @@ export interface DiscoveredEmulators {
  */
 export function getEmulatorHostAndPort(
   emulator: keyof DiscoveredEmulators,
-  conf?: EmulatorConfig,
+  endpoint?: HostAndPort,
```

Comment:
Isn't this a breaking API change?

Reply:
Yeah, I totally overlooked that when I did this. Should I revert this? I made this change because the old types were causing errors because of improved TS analysis, and the optional property was easier to work with.

Reply:
Yeah, I think you should revert it. We can only change the public API after an API council review, and a breaking change only on major version bumps.

What you could do is: propose a new method with the new signature, have the council approve that, then you implement it while deprecating this one. Then we would remove the older deprecated version in v12.

Reply:
Sounds good, I reverted it.

We used to throw an error if only one of `host` or `port` were defined, but this branch was unreachable because both properties are required, so TS reported an error. I had to remove the code that threw the error, along with the tests for it.

Reply:
I'd actually suggest just keeping the error branch but add `as any` or something to make TS okay with it. It's only unreachable if the caller is also using Typescript and their code passes type checking. But our developers may use plain JavaScript (e.g. with Node.js) and run into it.

Reply:
Sounds good, will do! Thanks for the input here- I wasn't really sure what was the best way forward with the error branch.